transaction-context/src/lib.rs

```diff
@@ -377,6 +377,37 @@ impl TransactionContext {
             .ok_or(InstructionError::CallDepth)
     }
 
+    /// Returns the index of this instruction in a 'stack' of instructions.
+    ///
+    /// Useful for displaying instruction index paths (ie. for the second CPI of the third top level
+    /// instruction, displaying 3.2).
+    ///
+    /// # Example
+    ///
+    /// Observe how the index resets every time a new top-level instruction is called.
+    ///
+    /// * #1 Program A (no index)
+    ///   * #1.1 CPI to Program B (index 0)
```

Comment:
Explorers typically 1-index inner instruction indexes, but in the code we zero index them, `None` meaning ‘this is not an inner instruction.’